xtask/src/hooks.rs

```diff
@@ -122,27 +136,125 @@ fn shellcheck(all: bool, msg_info: &mut MessageInfo) -> cross::Result<()> {
     Ok(())
 }
 
+fn parse_command(value: &str) -> cross::Result<Vec<String>> {
+    shell_words::split(value).wrap_err_with(|| format!("could not parse command of {}", value))
+}
+
+fn python_dir(metadata: &cross::CargoMetadata) -> PathBuf {
+    metadata.workspace_root.join("docker").join("android")
+}
+
+fn python_env(cmd: &mut Command, metadata: &cross::CargoMetadata) {
+    cmd.env("PYTHONDONTWRITEBYTECODE", "1");
+    cmd.env(
+        "PYTHONPYCACHEPREFIX",
+        metadata.target_directory.join("__pycache__"),
+    );
+}
+
+fn python_lint(flake8: Option<&str>, msg_info: &mut MessageInfo) -> cross::Result<()> {
+    let metadata = cargo_metadata(msg_info)?;
+    let args = flake8
+        .map(parse_command)
+        .unwrap_or_else(|| Ok(vec!["flake8".to_owned()]))?;
+    let mut cmd = Command::new(
+        args.get(0)
+            .ok_or_else(|| eyre::eyre!("empty string provided for flake8 command"))?,
+    );
+    cmd.args(&args[1..]);
+    python_env(&mut cmd, &metadata);
+    if msg_info.is_verbose() {
+        cmd.arg("--verbose");
+    }
+    cmd.current_dir(python_dir(&metadata));
+    cmd.run(msg_info, false)?;
+
+    Ok(())
+}
+
+fn python_test(tox: Option<&str>, msg_info: &mut MessageInfo) -> cross::Result<()> {
+    let metadata = cargo_metadata(msg_info)?;
+    let args = tox
+        .map(parse_command)
+        .unwrap_or_else(|| Ok(vec!["tox".to_owned()]))?;
+    let mut cmd = Command::new(
+        args.get(0)
+            .ok_or_else(|| eyre::eyre!("empty string provided for tox command"))?,
+    );
+    cmd.args(&args[1..]);
+    cmd.args(["-e", "py3"]);
+    python_env(&mut cmd, &metadata);
+    cmd.arg("--workdir");
+    cmd.arg(&metadata.target_directory);
+    if msg_info.is_verbose() {
+        cmd.arg("--verbose");
+    }
+    cmd.current_dir(python_dir(&metadata));
+    cmd.run(msg_info, false)?;
+
+    Ok(())
+}
+
 pub fn check(
-    Check { all, .. }: Check,
+    Check {
+        all,
+        python,
+        flake8,
+        ..
+    }: Check,
     toolchain: Option<&str>,
     msg_info: &mut MessageInfo,
 ) -> cross::Result<()> {
-    msg_info.info("Running rustfmt, clippy, and shellcheck checks.")?;
+    let mut checks = vec!["rustfmt", "clippy", "shellcheck"];
+    if python {
+        checks.push("python");
+    }
+    let join_to = checks.len() - 1;
+    let mut joined = checks[0..join_to].join(", ");
+    if join_to > 1 {
+        joined.push(',');
+    }
+    msg_info.info(format_args!(
+        "Running {joined} and {} checks.",
+        checks[join_to]
+    ))?;
```

Comment:
I think it's fine to drop the `, and`

I think it's fine to drop the `, and` here

```suggestion
    let mut checks = vec!["rustfmt", "clippy", "shellcheck"];
    if python {
        checks.push("python");
    }
    msg_info.info(format_args!(
        "Running {} checks.",
        checks.join(",")
    ))?;
```